imctools/io/tiffwriter.py

```diff
@@ -122,3 +125,19 @@ def get_xml(self, dtype=None):
 
         xml = omexml.to_xml()
         return xml
+
+def change_dtype(a, dtype):
+    if dtype.kind in ['i', 'u']:
+        dinf = np.iinfo(dtype)
+        mina = a.min()
+        maxa = a.max()
+        if mina < dinf.min:
+            a = a.copy()
```

Comment:
Why am I not using: `numpy.clip`? This would be C code and thus much faster
https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.clip.html#numpy.clip